pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -484,6 +491,53 @@ func (mp *MetaPatch) Validate() ([]metav1.StatusCause, bool) {
 	return causes, len(causes) == 0
 }
 
+// Counter is a Priority SortType
+const Counter string = "counter"
+
+// List is a Priority SortType
+const List string = "list"
+
+// Ascending a Priority Order where the smaller count is preferred.
+const Ascending string = "ascending"
+
+// Descending a Priority Order where the larger count is preferred.
+const Descending string = "descending"
+
+// Priority is a sorting option for GameServers with Counters or Lists based on the count or
+// number of items in a List.
+// SortType: Sort by a "counter" or a "list".
+// Key: The name of the Counter or List. If not found on the GameServer, has no impact.
+// Order: Sort by "ascending" or "descending". Default is "descending" so bigger count is preferred.
+// "ascending" would be smaller count is preferred.
+type Priority struct {
+	SortType string `json:"sortType"`
```

Comment:
> We can't (or shouldn't) use "Type" because it's a key word in Golang. Maybe "SortType" or "Order" or "OrderType"?

That is an exceptional point. 

`PriorityType` - since it's part of priority? Seems like an easy enough keyword escaping.

```suggestion
	PriorityType string `json:"type"`
```

Which does solve it for mapping - but maybe is just waiting for a footgun to go off ("type" as a json mapping)?

or

```suggestion
	PriorityType string `json:"priorityType"`
```

Either works for me.

I could work with `OrderType` as - since then it aligns with `Order` below it.

So I'd say either align with the parent `Priorities` or the neighbour element `Order`. Either works for me 👍🏻